virtual_rainforest/models/soil/soil_model.py

```diff
@@ -132,29 +147,122 @@ def spinup(self) -> None:
         """Placeholder function to spin up the soil model."""
 
     def update(self) -> None:
-        """Placeholder function to update the soil model."""
+        """Function to update the soil model."""
 
-        # Convert time step from seconds to days
-        dt = self.update_interval.astype("timedelta64[s]").astype(float) / (
-            60.0 * 60.0 * 24.0
-        )
-
-        carbon_pool_updates = self.carbon.calculate_soil_carbon_updates(
-            self.data,
-            self.data["pH"],
-            self.data["bulk_density"],
-            self.data["soil_moisture"],
-            self.data["soil_temperature"],
-            self.data["percent_clay"],
-            dt,
-        )
+        # Find carbon pool updates by integration
+        updated_carbon_pools = self.integrate_soil_model()
 
         # Update carbon pools (attributes and data object)
         # n.b. this also updates the data object automatically
-        self.carbon.update_soil_carbon_pools(self.data, carbon_pool_updates)
+        self.replace_soil_pools(updated_carbon_pools)
 
         # Finally increment timing
         self.next_update += self.update_interval
 
     def cleanup(self) -> None:
         """Placeholder function for soil model cleanup."""
```

Comment:
(I know this isn't part of this PR, but I've only just noticed it.)

Do all models have a `cleanup()` method? What's it needed for? You don't normally have to worry about manually cleaning up resources with Python.